hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/ContractUpdateHandler.java

```diff
@@ -67,9 +67,8 @@ public void preHandle(@NonNull final PreHandleContext context) {
             final var key = asHederaKey(op.adminKeyOrThrow());
             key.ifPresent(context::addToReqNonPayerKeys);
         }
-        // TODO This code isn't right, autoRenewAccountId may be null
-        if (op.hasAutoRenewAccountId() && !op.autoRenewAccountId().equals(AccountID.DEFAULT)) {
-            context.addNonPayerKey(op.autoRenewAccountId(), INVALID_AUTORENEW_ACCOUNT);
+        if (op.hasAutoRenewAccountId() && !op.autoRenewAccountIdOrThrow().equals(AccountID.DEFAULT)) {
+            context.addNonPayerKey(op.autoRenewAccountIdOrThrow(), INVALID_AUTORENEW_ACCOUNT);
```

Comment:
We have some warnings in this file and some other files like below
```
Parameter 'query' is never used
```
We can add `requireNonNull` to avoid those warnings.

Reply:
done!